### PR TITLE
Add applyconfiguration generator to code-generator script

### PR DESCRIPTION
**Pull request by astefanutti**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR adds the `applyconfiguration` generator to the code-generator script for external types / CRDs, so server-side apply configuration and client APIs can be generated with it.

```release-note
Adds the applyconfiguration generator to the code-generator script that generates server-side apply configuration and client APIs
```

### CODE REVIEW

**Comment by cici37** [comment]
/cc @apelisse 
You might be interested :)
/triage accepted

**Comment by stevekuznetsov** [comment]
cc @jbpetz

**Comment by stevekuznetsov** [comment]
cc @jpbetz

File: staging/src/k8s.io/code-generator/generate-groups.sh [hunk 1]
```diff
@@ -77,13 +77,25 @@ if [ "${GENS}" = "all" ] || grep -qw "deepcopy" <<<"${GENS}"; then
...
   echo "Generating clientset for ${GROUPS_WITH_VERSIONS} at ${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}"
+  if [ "${GENS}" = "all" ] || grep -qw "applyconfiguration" <<<"${GENS}"; then
+    APPLY_CONFIGURATION_PACKAGE="${OUTPUT_PKG}/${APPLYCONFIGURATION_PKG_NAME:-applyconfiguration}"
+  fi
   "${gobin}/client-gen" \
       --clientset-name "${CLIENTSET_NAME_VERSIONED:-versioned}" \
       --input-base "" \
       --input "$(codegen::join , "${FQ_APIS[@]}")" \
       --output-package "${OUTPUT_PKG}/${CLIENTSET_PKG_NAME:-clientset}" \
+      --apply-configuration-package "${APPLY_CONFIGURATION_PACKAGE:-}" \
```
Comment (jpbetz) [inline staging/src/k8s.io/code-generator/generate-groups.sh:98]:
```suggestion
      --apply-configuration-package "${APPLY_CONFIGURATION_PACKAGE:-applyconfiguration}" \
```

Here too?

Comment (astefanutti) [reply staging/src/k8s.io/code-generator/generate-groups.sh:98]: That one is a bit different from the others above. It corresponds to the package of apply configurations, previously generated by `applyconfiguration-gen`, and that's optional so Apply functions are not generated if empty, which happens if the `applyconfiguration` generator wasn't specified.

**Comment by ffromani** [comment]
/approve
changes to noderesourcetopology-api LGTM

**Review by swatisehgal — APPROVED** [review]
/approve

for the changes related to noderesourcetopology-api

**Pull request merged**